source-mysql/discovery.go

```diff
@@ -287,11 +294,35 @@ func (db *mysqlDatabase) translateRecordField(columnType interface{}, val interf
 	return val, nil
 }
 
-const queryDiscoverColumns = `
-  SELECT table_schema, table_name, ordinal_position, column_name, is_nullable, data_type, column_type
-  FROM information_schema.columns
+const queryDiscoverTables = `
+  SELECT table_schema, table_name, table_type
+  FROM information_schema.tables
   WHERE table_schema != 'information_schema' AND table_schema != 'performance_schema'
-    AND table_schema != 'mysql' AND table_schema != 'sys'
+    AND table_schema != 'mysql' AND table_schema != 'sys';`
+
+func getTables(ctx context.Context, conn *client.Conn) ([]*sqlcapture.DiscoveryInfo, error) {
+	var results, err = conn.Execute(queryDiscoverTables)
+	if err != nil {
+		return nil, fmt.Errorf("error listing tables: %w", err)
+	}
+	defer results.Close()
+
+	var tables []*sqlcapture.DiscoveryInfo
+	for _, row := range results.Values {
+		tables = append(tables, &sqlcapture.DiscoveryInfo{
+			Schema:    string(row[0].AsString()),
+			Name:      string(row[1].AsString()),
+			BaseTable: strings.EqualFold(string(row[2].AsString()), "BASE TABLE"),
+		})
+	}
+	return tables, nil
+}
+
+const queryDiscoverColumns = `
+  SELECT col.table_schema, col.table_name, col.ordinal_position, col.column_name, col.is_nullable, col.data_type, col.column_type, tbl.table_type
+  FROM information_schema.columns col
+  JOIN information_schema.tables tbl
+    ON (col.table_schema = tbl.table_schema AND col.table_name = tbl.table_name)
   ORDER BY table_schema, table_name, ordinal_position;`
 
 func getColumns(ctx context.Context, conn *client.Conn) ([]sqlcapture.ColumnInfo, error) {
```

Comment:
should ColumnInfo below be extended with BaseTable ?

Reply:
Ah, I forgot to revert my changes to this query after I decided that just factoring out table-discovery as its own distinct thing was cleaner. I'll revert these edits here.